sdk-libs/sdk/src/compressible/decompress_idempotent.rs

```diff
@@ -0,0 +1,199 @@
+use solana_account_info::AccountInfo;
+use solana_clock::Clock;
+use solana_cpi::invoke_signed;
+use solana_msg::msg;
+use solana_pubkey::Pubkey;
+use solana_rent::Rent;
+use solana_system_interface::instruction as system_instruction;
+use solana_sysvar::Sysvar;
+
+use crate::compressible::compress_pda::CompressionTiming;
+
+pub const COMPRESSION_DELAY: u64 = 100;
+
+/// Helper function to decompress a compressed account into a PDA idempotently.
+///
+/// This function is idempotent, meaning it can be called multiple times with the same compressed account
+/// and it will only decompress it once. If the PDA already exists and is initialized, it returns early.
+///
+/// # Arguments
+/// * `pda_account` - The PDA account to decompress into
+/// * `compressed_account` - The compressed account to decompress
+/// * `proof` - Validity proof
+/// * `cpi_accounts` - Accounts needed for CPI
+/// * `owner_program` - The program that will own the PDA
+/// * `rent_payer` - The account to pay for PDA rent
+/// * `system_program` - The system program
+///
+/// # Returns
+/// * `Ok(())` if the compressed account was decompressed successfully or PDA already exists
+/// * `Err(LightSdkError)` if there was an error
+pub fn decompress_idempotent<'info, A>(
+    pda_account: &AccountInfo<'info>,
+    compressed_account: LightAccount<'_, A>,
+    proof: ValidityProof,
+    cpi_accounts: CpiAccounts<'_, 'info>,
+    owner_program: &Pubkey,
+    rent_payer: &AccountInfo<'info>,
+    system_program: &AccountInfo<'info>,
+) -> Result<(), LightSdkError>
+where
+    A: DataHasher
+        + LightDiscriminator
+        + BorshSerialize
+        + BorshDeserialize
+        + Default
+        + Clone
+        + CompressionTiming,
+{
+    decompress_multiple_idempotent(
+        &[pda_account],
+        vec![compressed_account],
+        proof,
+        cpi_accounts,
+        owner_program,
+        rent_payer,
+        system_program,
+    )
+}
+
+/// Helper function to decompress multiple compressed accounts into PDAs idempotently.
+///
+/// This function is idempotent, meaning it can be called multiple times with the same compressed accounts
+/// and it will only decompress them once. If a PDA already exists and is initialized, it skips that account.
+///
+/// # Arguments
+/// * `pda_accounts` - The PDA accounts to decompress into
+/// * `compressed_accounts` - The compressed accounts to decompress
+/// * `proof` - Single validity proof for all accounts
+/// * `cpi_accounts` - Accounts needed for CPI
+/// * `owner_program` - The program that will own the PDAs
+/// * `rent_payer` - The account to pay for PDA rent
+/// * `system_program` - The system program
+///
+/// # Returns
+/// * `Ok(())` if all compressed accounts were decompressed successfully or PDAs already exist
+/// * `Err(LightSdkError)` if there was an error
+pub fn decompress_multiple_idempotent<'info, A>(
+    pda_accounts: &[&AccountInfo<'info>],
+    compressed_accounts: Vec<LightAccount<'_, A>>,
```

Comment:
```suggestion
    compressed_accounts: &[LightAccount<'_, A>],
```

We should avoid vectors.